justfile

```diff
@@ -1,6 +1,82 @@
-# Setup the tools needed to develop
-setup-tools:
-    cargo install cargo-release insta 
-# publish rust crates    
-release-crates:
+#!/usr/bin/env -S just --justfile
+
+# Set shell configurations
+set windows-shell := ["powershell"]
+set shell := ["bash", "-cu"]
+
+# Default target: List all tasks with updated information
+_default:
+    just --list -u
+
+# Setup environment for Rust and Node.js
+setup:
+  # Install Rust-related tools
+  cargo install cargo-binstall
+  cargo binstall taplo-cli cargo-release cargo-insta cargo-deny -y
+  
+  # Setup Node.js environment
+  corepack enable
+  pnpm install
+  
+  @echo '✅ Setup complete!'
+
+# Check readiness of the project
+ready:
+  just fmt  
+  just check  
+  just lint 
+  just test
+  @echo '✅ All passed!'
+
+# Publish Rust crates to crates.io
+release:
```

Comment:
release-crates or release-rust? since we have different ways to release javascript and rust

Reply:
`release-rust` may be more intuitive and convenient.